src/query/ts/m3db/encoded_unconsolidated_series_iterator.go

```diff
@@ -76,7 +81,7 @@ func (it *encodedSeriesIterUnconsolidated) Next() bool {
 		dp, _, _ := iter.Current()
 		values = append(values,
 			xts.Datapoint{
-				Timestamp: dp.Timestamp,
+				Timestamp: dp.Timestamp.Add(it.offset),
```

Comment:
Hm yeah, it feels weird that we do this at this layer and also in the query/storage layer.

Is there any possibility that the `Offset` function itself could just adjust the timestamps back to what is expected?